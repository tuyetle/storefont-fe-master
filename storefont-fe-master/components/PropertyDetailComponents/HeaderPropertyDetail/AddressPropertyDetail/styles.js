import { css } from 'glamor';
import variables from '~/styles/variables';

export default {
  addressTransaction: css({
    display: 'block',
  }),

  label: css({
    marginLeft: '0',
    marginRight: '1em',
    color: variables.$listingItemTitleColor,
    fontSize: '0.6875em',

    '@media(min-width: 768px)': {
      fontSize: '1.125em',
    },

  }),

  transactionLabel: css({
    fontSize: '0.5625em',
    textTransform: 'uppercase',
    border: `1px solid ${variables.$locationTextColor}`,
    padding: '0.3125em 0.5625em 0.2125em',

    '@media(min-width: 768px)': {
      fontSize: '0.9375em',
    },
  }),

  iconLocation: css({
    width: '0.6em',
    height: '0.7em',
    display: 'inline-block',
    verticalAlign: 'middle',
    marginRight: '0.2em',


    '@media(min-width: 768px)': {
      width: '1.2em',
      height: '1.5em',
      marginRight: '0.4em',
      verticalAlign: 'top',
    },
  }),

};

