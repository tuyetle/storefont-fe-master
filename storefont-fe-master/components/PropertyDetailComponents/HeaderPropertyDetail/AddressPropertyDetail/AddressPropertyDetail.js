import React from 'react';
import PropTypes from 'prop-types';
import LocationIcon from '~/static/assets/img/ic-location.svg';
import styles from './styles';

const AddressPropertyDetail = ({ transactionType, address }) => (
  <div className={styles.addressTransaction}>
    <LocationIcon className={styles.iconLocation} />
    <span className={styles.label}>{address}</span>
    <span className={styles.transactionLabel}>{transactionType}</span>
  </div>
);

AddressPropertyDetail.propTypes = {
  address: PropTypes.string.isRequired,
  transactionType: PropTypes.string.isRequired,
};

export default AddressPropertyDetail;
