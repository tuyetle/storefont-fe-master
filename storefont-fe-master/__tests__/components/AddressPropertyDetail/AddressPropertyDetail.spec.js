import React from 'react';
import renderer from 'react-test-renderer';
import AddressPropertyDetail from '~/components/PropertyDetailComponents/HeaderPropertyDetail/AddressPropertyDetail/AddressPropertyDetail';

describe('(Component) AddressPropertyDetail', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      address: 'An Dương Vương, Phường Phú Thượng, Quận Tây Hồ',
      transactionType: 'Bán',
    };
    component = <AddressPropertyDetail {...props} />;
  });

  it('should renders correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

