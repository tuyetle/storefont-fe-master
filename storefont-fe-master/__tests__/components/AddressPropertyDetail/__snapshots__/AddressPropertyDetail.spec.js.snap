// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AddressPropertyDetail should renders correctly 1`] = `
<div
  className={
    Object {
      "data-css-fisw11": "",
    }
  }
>
  <svg
    className={
      Object {
        "data-css-yhgus": "",
      }
    }
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="#686868"
    >
      <path
        d="M255.5 111.5c-35.2 0-63.8 28.6-63.8 63.8 0 35.2 28.6 63.8 63.8 63.8 35.1 0 63.8-28.6 63.8-63.8 0-35.2-28.6-63.8-63.8-63.8zm0 109.3c-25.1 0-45.6-20.4-45.6-45.6 0-25.1 20.4-45.6 45.6-45.6 25.1 0 45.6 20.5 45.6 45.6 0 25.2-20.5 45.6-45.6 45.6zm0 0"
      />
      <path
        d="M377.5 53.3c-33-33-76.9-51.2-123.5-51.2-46.7 0-90.6 18.2-123.6 51.2-61 61.1-68.6 175.9-16.4 245.6L254 501l139.7-201.9c52.4-69.8 44.8-184.7-16.2-245.8zm1.4 235.2L253.9 469 128.7 288.3c-47.3-63.2-40.6-166.9 14.5-222 29.6-29.5 68.9-45.8 110.7-45.8S335 36.8 364.5 66.3c55.2 55.1 62 158.8 14.4 222.2zm0 0"
      />
    </g>
  </svg>
  <span
    className={
      Object {
        "data-css-1awbpa": "",
      }
    }
  >
    An Dương Vương, Phường Phú Thượng, Quận Tây Hồ
  </span>
  <span
    className={
      Object {
        "data-css-vfobqd": "",
      }
    }
  >
    Bán
  </span>
</div>
`;
